jetson/percep/artag_detector.cpp

```diff
@@ -16,8 +15,14 @@ void onMouse(int event, int x, int y, int flags, void *userdata) {
     }
 }
 
-TagDetector::TagDetector() {  //initializes detector object with pre-generated dictionary of tags
-
+TagDetector::TagDetector(const rapidjson::Document &mRoverConfig) //initializes detector object with pre-generated dictionary of tags
+    :  //Populate Constants from Config File
+   BUFFER_ITERATIONS{mRoverConfig["buffer_iterations"].GetInt()},
+   MARKER_BORDER_BITS{mRoverConfig["alvar_params"]["markerBorderBits"].GetInt()},
+   DO_CORNER_REFINEMENT{!!mRoverConfig["alvar_params"]["doCornerRefinement"].GetInt()},
+   POLYGONAL_APPROX_ACCURACY_RATE{mRoverConfig["alvar_params"]["polygonalApproxAccuracyRate"].GetDouble()},
+   MM_PER_M{mRoverConfig["mm_per_m"].GetInt()},
+   DEFAULT_TAG_VAL{mRoverConfig["DEFAULT_TAG_VAL"].GetInt()} {
     cv::FileStorage fsr("jetson/percep/alvar_dict.yml", cv::FileStorage::READ);
```

Comment:
Can we format this better? Maybe move the colon up a line, make the comment on line 19 in line with all of the variable declarations, and add a new line after the open curly bracket on line 25?

Reply:
Checkout how we did the formatting of the constructor in PCL.cpp. That formatting looks better